test/e2e/service_to_service_test.go

```diff
@@ -153,11 +153,8 @@ func testProxyToHelloworld(t *testing.T, clients *test.Clients, helloworldURL *u
 	if err != nil {
 		t.Fatal("Unexpected error when sending request to helloworld:", err)
 	}
-	expectedStatus := http.StatusNotFound
-	if accessibleExternal {
-		expectedStatus = http.StatusOK
-	}
-	if got, want := response.StatusCode, expectedStatus; got != want {
+
+	if got, want := response.StatusCode, http.StatusOK; got != want {
```

Comment:
Can we skip the test much sooner?

ie. in `TestSvcToSvcViaActivator` since we have the information to know if the test would succeed before creating the Knative Service etc.